File: admin/apps/web-antd/src/api/admin/content-production.ts
import { requestClient } from '#/api/request';

export interface AgentConfig {
  agent: string;
  provider: string;
  baseUrl: string;
  model: string;
  timeoutSeconds: number;
  keyConfigured: boolean;
  enabled: boolean;
}

export interface ContentScript {
  id: number;
  merchantId: number;
  merchantName: string;
  topicId: number;
  topicTitle: string;
  title: string;
  opening: string;
  body: string;
  ending: string;
  cta: string;
  fullScript: string;
  shootingNotesJson: string;
  shootingNotes: string[];
  status: string;
  inputSnapshot: string;
  resultJson: string;
  errorMessage: string;
  createdAt: string;
  updatedAt: string;
}

export interface ContentScriptListResult {
  list: ContentScript[];
  total: number;
  page: number;
  size: number;
}

export interface StoryboardShot {
  index: number;
  duration: string;
  location: string;
  camera: string;
  content: string;
  line: string;
  note: string;
}

export interface ContentStoryboard {
  id: number;
  merchantId: number;
  merchantName: string;
  topicId: number;
  topicTitle: string;
  scriptId: number;
  scriptTitle: string;
  shotsJson: string;
  shots: StoryboardShot[];
  status: string;
  inputSnapshot: string;
  resultJson: string;
  errorMessage: string;
  createdAt: string;
  updatedAt: string;
}

export interface ContentStoryboardListResult {
  list: ContentStoryboard[];
  total: number;
  page: number;
  size: number;
}

export interface ShootingTask {
  id: number;
  merchantId: number;
  merchantName: string;
  topicId: number;
  topicTitle: string;
  scriptId: number;
  scriptTitle: string;
  storyboardId: number;
  taskTitle: string;
  shotCount: number;
  shotsJson: string;
  shots: StoryboardShot[];
  assignee: string;
  shootTime?: string;
  deadline?: string;
  status: string;
  materialUrl: string;
  remark: string;
  createdAt: string;
  updatedAt: string;
}

export interface ShootingTaskListResult {
  list: ShootingTask[];
  total: number;
  page: number;
  size: number;
}

export interface PublishSchedule {
  id: number;
  merchantId: number;
  merchantName: string;
  topicId: number;
  topicTitle: string;
  scriptId: number;
  scriptTitle: string;
  storyboardId: number;
  videoTitle: string;
  publishTime?: string;
  owner: string;
  douyinAccount: string;
  materialStatus: string;
  status: string;
  remark: string;
  createdAt: string;
  updatedAt: string;
}

export interface PublishScheduleListResult {
  list: PublishSchedule[];
  total: number;
  page: number;
  size: number;
}

export interface ContentReviewTask {
  id: number;
  merchantId: number;
  merchantName: string;
  scheduleId: number;
  videoTitle: string;
  status: string;
  resultJson: string;
  errorMessage: string;
  periodStart?: string;
  periodEnd?: string;
  playCount: number;
  likeCount: number;
  commentCount: number;
  shareCount: number;
  dealCount: number;
  writeOffAmount: number;
  result?: {
    conclusion?: string;
    nextTopics?: string[];
    optimizes?: string[];
    summary?: string;
    suggestions?: string[];
  };
  createdAt: string;
  updatedAt: string;
}

export function getContentScripts(params: {
  keyword?: string;
  merchantId?: number;
  page?: number;
  size?: number;
  status?: string;
  topicId?: number;
}) {
  return requestClient.get<ContentScriptListResult>('/scripts', { params });
}

export function generateContentScript(data: {
  extraRequirement?: string;
  topicId: number;
}) {
  return requestClient.post<ContentScript>('/scripts/generate', data, {
    timeout: 120_000,
  });
}

export function updateContentScriptStatus(id: number, status: string) {
  return requestClient.put<ContentScript>(`/scripts/${id}/status`, { status });
}

export function getContentStoryboards(params: {
  keyword?: string;
  merchantId?: number;
  page?: number;
  scriptId?: number;
  size?: number;
  status?: string;
}) {
  return requestClient.get<ContentStoryboardListResult>('/storyboards', {
    params,
  });
}

export function generateContentStoryboard(data: {
  locations?: string[];
  scriptId: number;
}) {
  return requestClient.post<ContentStoryboard>('/storyboards/generate', data, {
    timeout: 120_000,
  });
}

export function updateContentStoryboardStatus(id: number, status: string) {
  return requestClient.put<ContentStoryboard>(`/storyboards/${id}/status`, {
    status,
  });
}

export function getShootingTasks(params: {
  keyword?: string;
  merchantId?: number;
  page?: number;
  size?: number;
  status?: string;
  storyboardId?: number;
}) {
  return requestClient.get<ShootingTaskListResult>('/shooting-tasks', {
    params,
  });
}

export function createShootingTask(data: {
  assignee?: string;
  deadline?: string;
  materialUrl?: string;
  remark?: string;
  shootTime?: string;
  status?: string;
  storyboardId: number;
  taskTitle?: string;
}) {
  return requestClient.post<ShootingTask>('/shooting-tasks', data);
}

export function updateShootingTaskStatus(
  id: number,
  data: { materialUrl?: string; remark?: string; status?: string },
) {
  return requestClient.put<ShootingTask>(`/shooting-tasks/${id}/status`, data);
}

export function getPublishSchedules(params: {
  keyword?: string;
  merchantId?: number;
  page?: number;
  size?: number;
  status?: string;
  storyboardId?: number;
}) {
  return requestClient.get<PublishScheduleListResult>('/schedules', {
    params,
  });
}

export function createPublishSchedule(data: Partial<PublishSchedule>) {
  return requestClient.post<PublishSchedule>('/schedules', data);
}

export function updatePublishSchedule(
  id: number,
  data: Partial<PublishSchedule>,
) {
  return requestClient.put<PublishSchedule>(`/schedules/${id}`, data);
}

export function updatePublishScheduleStatus(
  id: number,
  data: { materialStatus?: string; remark?: string; status?: string },
) {
  return requestClient.put<PublishSchedule>(`/schedules/${id}/status`, data);
}

export function getContentReviews(params: {
  merchantId?: number;
  page?: number;
  size?: number;
}) {
  return requestClient.get<{
    list: ContentReviewTask[];
    total: number;
    page: number;
    size: number;
  }>('/reviews', { params });
}

export function generateContentReview(data: {
  commentCount?: number;
  dealCount?: number;
  likeCount?: number;
  periodEnd?: string;
  periodStart?: string;
  playCount?: number;
  scheduleId: number;
  shareCount?: number;
  writeOffAmount?: number;
}) {
  return requestClient.post<ContentReviewTask>('/reviews/generate', data, {
    timeout: 120_000,
  });
}

export function getAgentConfigs() {
  return requestClient.get<AgentConfig[]>('/agent-configs');
}

File: server/internal/model/content_production.go
package model

import "time"

const (
	ContentScriptStatusDraft     = "草稿"
	ContentScriptStatusConfirmed = "已确认"
	ContentScriptStatusDisabled  = "停用"

	ContentStoryboardStatusDraft     = "草稿"
	ContentStoryboardStatusConfirmed = "已确认"

	ShootingTaskStatusPending  = "待拍摄"
	ShootingTaskStatusShooting = "拍摄中"
	ShootingTaskStatusShot     = "已拍摄"
	ShootingTaskStatusEdited   = "已剪辑"
	ShootingTaskStatusDone     = "已完成"

	PublishScheduleStatusPending   = "待发布"
	PublishScheduleStatusPublished = "已发布"
	PublishScheduleStatusReviewed  = "已复盘"

	ContentReviewStatusCompleted = "已完成"
	ContentReviewStatusFailed    = "失败"
)

type ContentScript struct {
	ID                uint64    `gorm:"primaryKey" json:"id"`
	MerchantID        uint64    `gorm:"index;not null" json:"merchantId"`
	MerchantName      string    `gorm:"size:128;not null;default:''" json:"merchantName"`
	TopicID           uint64    `gorm:"index;not null;default:0" json:"topicId"`
	TopicTitle        string    `gorm:"size:255;not null;default:''" json:"topicTitle"`
	Title             string    `gorm:"size:255;not null;index" json:"title"`
	Opening           string    `gorm:"size:1000;not null;default:''" json:"opening"`
	Body              string    `gorm:"type:text" json:"body"`
	Ending            string    `gorm:"size:1000;not null;default:''" json:"ending"`
	CTA               string    `gorm:"size:1000;not null;default:''" json:"cta"`
	FullScript        string    `gorm:"type:text" json:"fullScript"`
	ShootingNotesJSON string    `gorm:"type:text" json:"shootingNotesJson"`
	Status            string    `gorm:"size:32;not null;default:'草稿'" json:"status"`
	InputSnapshot     string    `gorm:"type:text" json:"inputSnapshot"`
	ResultJSON        string    `gorm:"type:text" json:"resultJson"`
	ErrorMessage      string    `gorm:"size:1000;not null;default:''" json:"errorMessage"`
	CreatedBy         uint64    `gorm:"index;not null;default:0" json:"createdBy"`
	UpdatedBy         uint64    `gorm:"index;not null;default:0" json:"updatedBy"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
}

type ContentStoryboard struct {
	ID            uint64    `gorm:"primaryKey" json:"id"`
	MerchantID    uint64    `gorm:"index;not null" json:"merchantId"`
	MerchantName  string    `gorm:"size:128;not null;default:''" json:"merchantName"`
	TopicID       uint64    `gorm:"index;not null;default:0" json:"topicId"`
	TopicTitle    string    `gorm:"size:255;not null;default:''" json:"topicTitle"`
	ScriptID      uint64    `gorm:"index;not null;default:0" json:"scriptId"`
	ScriptTitle   string    `gorm:"size:255;not null;default:''" json:"scriptTitle"`
	ShotsJSON     string    `gorm:"type:text" json:"shotsJson"`
	Status        string    `gorm:"size:32;not null;default:'草稿'" json:"status"`
	InputSnapshot string    `gorm:"type:text" json:"inputSnapshot"`
	ResultJSON    string    `gorm:"type:text" json:"resultJson"`
	ErrorMessage  string    `gorm:"size:1000;not null;default:''" json:"errorMessage"`
	CreatedBy     uint64    `gorm:"index;not null;default:0" json:"createdBy"`
	UpdatedBy     uint64    `gorm:"index;not null;default:0" json:"updatedBy"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

type PublishSchedule struct {
	ID             uint64     `gorm:"primaryKey" json:"id"`
	MerchantID     uint64     `gorm:"index;not null" json:"merchantId"`
	MerchantName   string     `gorm:"size:128;not null;default:''" json:"merchantName"`
	TopicID        uint64     `gorm:"index;not null;default:0" json:"topicId"`
	TopicTitle     string     `gorm:"size:255;not null;default:''" json:"topicTitle"`
	ScriptID       uint64     `gorm:"index;not null;default:0" json:"scriptId"`
	ScriptTitle    string     `gorm:"size:255;not null;default:''" json:"scriptTitle"`
	StoryboardID   uint64     `gorm:"index;not null;default:0" json:"storyboardId"`
	VideoTitle     string     `gorm:"size:255;not null;index" json:"videoTitle"`
	PublishTime    *time.Time `json:"publishTime"`
	Owner          string     `gorm:"size:64;not null;default:''" json:"owner"`
	DouyinAccount  string     `gorm:"size:128;not null;default:''" json:"douyinAccount"`
	MaterialStatus string     `gorm:"size:64;not null;default:'待拍摄'" json:"materialStatus"`
	Status         string     `gorm:"size:32;not null;default:'待发布'" json:"status"`
	Remark         string     `gorm:"size:1000;not null;default:''" json:"remark"`
	CreatedBy      uint64     `gorm:"index;not null;default:0" json:"createdBy"`
	UpdatedBy      uint64     `gorm:"index;not null;default:0" json:"updatedBy"`
	CreatedAt      time.Time  `json:"createdAt"`
	UpdatedAt      time.Time  `json:"updatedAt"`
}

type ShootingTask struct {
	ID           uint64     `gorm:"primaryKey" json:"id"`
	MerchantID   uint64     `gorm:"index;not null" json:"merchantId"`
	MerchantName string     `gorm:"size:128;not null;default:''" json:"merchantName"`
	TopicID      uint64     `gorm:"index;not null;default:0" json:"topicId"`
	TopicTitle   string     `gorm:"size:255;not null;default:''" json:"topicTitle"`
	ScriptID     uint64     `gorm:"index;not null;default:0" json:"scriptId"`
	ScriptTitle  string     `gorm:"size:255;not null;default:''" json:"scriptTitle"`
	StoryboardID uint64     `gorm:"index;not null;default:0" json:"storyboardId"`
	TaskTitle    string     `gorm:"size:255;not null;index" json:"taskTitle"`
	ShotCount    int        `gorm:"not null;default:0" json:"shotCount"`
	ShotsJSON    string     `gorm:"type:text" json:"shotsJson"`
	Assignee     string     `gorm:"size:64;not null;default:''" json:"assignee"`
	ShootTime    *time.Time `json:"shootTime"`
	Deadline     *time.Time `json:"deadline"`
	Status       string     `gorm:"size:32;not null;default:'待拍摄'" json:"status"`
	MaterialURL  string     `gorm:"size:500;not null;default:''" json:"materialUrl"`
	Remark       string     `gorm:"size:1000;not null;default:''" json:"remark"`
	CreatedBy    uint64     `gorm:"index;not null;default:0" json:"createdBy"`
	UpdatedBy    uint64     `gorm:"index;not null;default:0" json:"updatedBy"`
	CreatedAt    time.Time  `json:"createdAt"`
	UpdatedAt    time.Time  `json:"updatedAt"`
}

type ContentReviewTask struct {
	ID             uint64     `gorm:"primaryKey" json:"id"`
	MerchantID     uint64     `gorm:"index;not null" json:"merchantId"`
	MerchantName   string     `gorm:"size:128;not null;default:''" json:"merchantName"`
	ScheduleID     uint64     `gorm:"index;not null;default:0" json:"scheduleId"`
	VideoTitle     string     `gorm:"size:255;not null;default:''" json:"videoTitle"`
	PeriodStart    *time.Time `json:"periodStart"`
	PeriodEnd      *time.Time `json:"periodEnd"`
	Status         string     `gorm:"size:32;not null;default:'已完成'" json:"status"`
	InputSnapshot  string     `gorm:"type:text" json:"inputSnapshot"`
	ResultJSON     string     `gorm:"type:text" json:"resultJson"`
	ErrorMessage   string     `gorm:"size:1000;not null;default:''" json:"errorMessage"`
	PlayCount      int64      `gorm:"not null;default:0" json:"playCount"`
	LikeCount      int64      `gorm:"not null;default:0" json:"likeCount"`
	CommentCount   int64      `gorm:"not null;default:0" json:"commentCount"`
	ShareCount     int64      `gorm:"not null;default:0" json:"shareCount"`
	DealCount      int64      `gorm:"not null;default:0" json:"dealCount"`
	WriteOffAmount float64    `gorm:"not null;default:0" json:"writeOffAmount"`
	CreatedBy      uint64     `gorm:"index;not null;default:0" json:"createdBy"`
	UpdatedBy      uint64     `gorm:"index;not null;default:0" json:"updatedBy"`
	CreatedAt      time.Time  `json:"createdAt"`
	UpdatedAt      time.Time  `json:"updatedAt"`
}
